golem-worker-service-base/tests/api_oas_convert_tests.rs

```diff
@@ -0,0 +1,960 @@
+        "routes": [
+            {
+                "method": "Post",
+                "path": "/v0.1.0/products",
+                "binding": {
+                    "bindingType": "default",
+                    "component": {
+                        "name": "product-service",
+                        "version": 0
+                    },
+                    "response": "let product = request.body; {status: 201u64, body: {success: true, id: product.id}}",
+                    "responseMappingInput": {
+                        "types": {
+                            "request": {
+                                "fields": [
+                                    {
+                                        "name": "body",
+                                        "typ": {
+                                            "fields": [
+                                                {
+                                                    "name": "id",
+                                                    "typ": {
+                                                        "type": "Str"
+                                                    }
+                                                },
+                                                {
+                                                    "name": "name",
+                                                    "typ": {
+                                                        "type": "Str"
+                                                    }
+                                                },
+                                                {
+                                                    "name": "price",
+                                                    "typ": {
+                                                        "type": "F32"
+                                                    }
+                                                }
+                                            ],
+                                            "type": "Record"
+                                        }
+                                    }
+                                ],
+                                "type": "Record"
+                            }
+                        }
+                    }
+                }
+            }
+        ]
+    }"#;
+
+    // Parse the JSON
+    let response_data: HttpApiDefinitionResponseData = from_str(json_data).unwrap();
+
+    // Convert to OpenAPI
+    let openapi_response =
+        OpenApiHttpApiDefinitionResponse::from_http_api_definition_response_data(&response_data)
+            .unwrap();
+
+    // Verify the OpenAPI YAML contains the correct request body schema
+    let yaml_str = openapi_response.openapi_yaml;
+
+    // Check request body schema
+    assert!(yaml_str.contains("requestBody:"));
+    assert!(yaml_str.contains("required: true"));
+    assert!(yaml_str.contains("application/json:"));
+    assert!(yaml_str.contains("type: object"));
```

Comment:
Junk test.

You should explicitly pull out the type schema for request and response and verify that it's as expected (you could do JSON normalization to make testing easier).